entrypoint.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+set -e
+while getopts "a:b:c:d:e:f:g:h:i:j:" o; do
+   case "${o}" in
+       a)
+         export scanType=${OPTARG}
+       ;;
+       b)
+         export format=${OPTARG}
+       ;;
+       c)
+         export template=${OPTARG}
+       ;;
+       d)
+         export exitCode=${OPTARG}
+       ;;
+       e)
+         export ignoreUnfixed=${OPTARG}
+       ;;
+       f)
+         export vulnType=${OPTARG}
+       ;;
+       g)
+         export severity=${OPTARG}
+       ;;
+       h)
+         export output=${OPTARG}
+       ;;
+       i)
+         export imageRef=${OPTARG}
+       ;;
+       j)
+         export scanRef=${OPTARG}
+       ;;
+  esac
+done
+
+if [ $scanRef ];then
+  imageRef=$scanRef
+fi
+
+if [ $ignoreUnfixed == "true" ];then
+  trivy ${scanType} \
+    --format ${format} \
+    --template ${template} \
+    --exit-code ${exitCode} \
+    --ignore-unfixed \
+    --vuln-type ${vulnType} \
+    --severity ${severity} \
+    --output ${output} \
+    ${imageRef}
+else
+  trivy ${scanType} \
+    --format ${format} \
+    --template ${template} \
+    --exit-code ${exitCode} \
+    --vuln-type ${vulnType} \
+    --severity ${severity} \
+    --output ${output} \
+    ${imageRef}
+fi
```

Comment:
This approach seems a bit unmaintainable to me in the long run. We have other boolean based options in Trivy that we may decide to add in the future (or any combination of them). Any way we can handle this better?

Reply:
I agree with @simar7 . Not sure if I understood the script / intend of this PR correctly, but I think we can keep this this action as pure declarative config in action.yaml.

If we want to run Trivy action for images and directories we have to use different subcommands:
- `trivy image <image ref>`
- `trivy fs <fs path>`

The above can be written down as:
- `trivy <artifact type> <artifact ref>`

where `<artifact type>` = `image | fs`
and `<artifact ref>` can be an image reference (e.g. `docker.io/library/nginx:1.16`) or a file system path (e.g. `/var/lib/my-app`)